src/production.rs

```diff
@@ -101,27 +103,37 @@ impl FromStr for Production {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub struct Iter<'a> {
-    iterator: slice::Iter<'a, Expression>,
+    slice: &'a [Expression],
 }
 
 impl<'a> Iterator for Iter<'a> {
     type Item = &'a Expression;
 
     fn next(&mut self) -> Option<Self::Item> {
-        self.iterator.next()
+        self.slice.split_first().map(|(first, rest)| {
+            self.slice = rest;
+            first
+        })
     }
 }
 
+#[derive(Debug, PartialEq, Eq)]
 pub struct IterMut<'a> {
-    iterator: slice::IterMut<'a, Expression>,
+    slice: &'a mut [Expression],
 }
 
 impl<'a> Iterator for IterMut<'a> {
     type Item = &'a mut Expression;
 
     fn next(&mut self) -> Option<Self::Item> {
-        self.iterator.next()
+        let slice = std::mem::take(&mut self.slice);
```

Comment:
workaround to satisfy borrow checker, as explained in [the Rustonomicon](https://doc.rust-lang.org/nomicon/borrow-splitting.html)